x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsProvider.java

```diff
@@ -282,9 +282,16 @@ public void createJobResultIndex(Job job, ClusterState state, final ActionListen
         final String indexName = tempIndexName;
 
         ActionListener<Boolean> indexAndMappingsListener = ActionListener.wrap(success -> {
-            final IndicesAliasesRequest request = client.admin().indices().prepareAliases()
-                    .addAlias(indexName, readAliasName, QueryBuilders.termQuery(Job.ID.getPreferredName(), job.getId()))
-                    .addAlias(indexName, writeAliasName).request();
+            final IndicesAliasesRequest request =
+                client.admin().indices().prepareAliases()
+                    .addAliasAction(
+                        IndicesAliasesRequest.AliasActions.add()
+                            .index(indexName)
+                            .alias(readAliasName)
+                            .isHidden(true)
```

Comment:
If the cluster has been upgraded then this might be adding a hidden alias on a non-hidden index. Is that a problem? (Maybe not - as long as you’re sure I’m good to merge this.)

For example, suppose a user runs ML in 7.6.1 and so has a non-hidden `.ml-anomalies-shared` index. Then they upgrade to 7.7 and create another job that uses the shared results index.

(This also makes me realise that before 7.last we’ll need another PR that makes pre-existing indices and aliases hidden. But that’s not critical for 7.7.)

Reply:
> If the cluster has been upgraded then this might be adding a hidden alias on a non-hidden index. Is that a problem? (Maybe not - as long as you’re sure I’m good to merge this.)

I've verified that in 7.7 hidden alias can point at a non-hidden index so it's not a problem.
Gordon confirmed that all four combinations (`{hidden, visible} x {index, alias}`) should work fine so there should not be any BWC issues while upgrading from 7.6 to 7.7.

> This also makes me realise that before 7.last we’ll need another PR that makes pre-existing indices and aliases hidden. But that’s not critical for 7.7.

I've just created tracking issue for that: https://github.com/elastic/elasticsearch/issues/53674